grid_map_core/src/Polygon.cpp

```diff
@@ -83,4 +87,98 @@ void Polygon::resetTimestamp()
   timestamp_ = 0.0;
 }
 
+Polygon Polygon::convexHullCircles(const Position center1,
+                                   const Position center2, const double radius,
+                                   const int nVertices)
+{
+  Eigen::Vector2d centerToVertex, centerToVertexTemp;
+  centerToVertex = center2 - center1;
+  centerToVertex.normalize();
+  centerToVertex *= radius;
+
+  grid_map::Polygon polygon;
+  for (int j = 0; j < ceil(nVertices / 2.0); j++) {
+    double theta = M_PI_2 + j * M_PI / (ceil(nVertices / 2.0) - 1);
+    Eigen::Rotation2D<double> rot2d(theta);
+    centerToVertexTemp = rot2d.toRotationMatrix() * centerToVertex;
+    polygon.addVertex(center1 + centerToVertexTemp);
+  }
+  for (int j = 0; j < ceil(nVertices / 2.0); j++) {
+    double theta = 3 * M_PI_2 + j * M_PI / (ceil(nVertices / 2.0) - 1);
+    Eigen::Rotation2D<double> rot2d(theta);
+    centerToVertexTemp = rot2d.toRotationMatrix() * centerToVertex;
+    polygon.addVertex(center2 + centerToVertexTemp);
+  }
+  return polygon;
+}
+
+Polygon Polygon::convexHullCircle(const Position center, const double radius,
+                                  const int nVertices)
+{
+  Eigen::Vector2d centerToVertex(radius, 0.0), centerToVertexTemp;
+
+  grid_map::Polygon polygon;
+  for (int j = 0; j < nVertices; j++) {
+    double theta = j * 2 * M_PI / (nVertices - 1);
+    Eigen::Rotation2D<double> rot2d(theta);
+    centerToVertexTemp = rot2d.toRotationMatrix() * centerToVertex;
+    polygon.addVertex(center + centerToVertexTemp);
+  }
+  return polygon;
+}
+
+Polygon Polygon::convexHull(Polygon& polygon1, Polygon& polygon2)
+{
+  std::vector<Position> vertices1 = polygon1.getVertices();
```

Comment:
Why do you copy the vertices here? You can either not copy or use a (maybe const) reference to them.